### PR TITLE
Create a view for std_vector input

### PR DESCRIPTION
**Pull request by Prasanna-Ponnusamy**
**_Describe your changes here:_**

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

File: src/t8_data/t8_containers.h [hunk 1]
```diff
@@ -31,6 +31,8 @@
 #include <t8.h>
 #include <t8_element.h>
 
+
```
Comment (Davknapp) [inline src/t8_data/t8_containers.h:34]: Can you delete these added lines please?

File: src/t8_data/t8_stdvector_conversion.hxx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * Basic conversion routines for std::vector to t8code data types.
+ */
+
+
+#ifndef T8_STDVECTOR_CONVERSION_H
+#define T8_STDVECTOR_CONVERSION_H
+
+#include <vector>
+
+/* TODO: write documentation */
```
Comment (Davknapp) [inline src/t8_data/t8_stdvector_conversion.hxx:33]: Please take care of this TODO

File: src/t8_data/t8_stdvector_conversion.hxx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+template <typename T>
+sc_array_t*
+t8_create_sc_array_view_from_vector (const std::vector<T> &vector)
+{
+  void *vector_data = (void *) vector.data ();
+  sc_array_t *new_view = sc_array_new_data (vector_data, sizeof (T), vector.size ());
+  return new_view;
+}
+
+template <typename T>
```
Comment (Davknapp) [inline src/t8_data/t8_stdvector_conversion.hxx:43]: Please document this function

File: src/t8_forest/t8_forest_partition.cxx [hunk 4]
```diff
@@ -26,6 +26,7 @@
 #include <t8_forest/t8_forest_general.h>
 #include <t8_cmesh/t8_cmesh_offset.h>
 #include <t8_element_cxx.hxx>
+#include <t8_data/t8_stdvector_conversion.hxx>
```
Comment (Davknapp) [inline src/t8_forest/t8_forest_partition.cxx:29]: You include the header but don't use it. Do you plan on using it?

**Review by Davknapp — CHANGES_REQUESTED** [review]
Nice addition! Can you also implement a test for the conversion? I suggest to create an sc_array from a vector, iterate over both and compare their values. Furthermore you can check other properties like the length.

File: src/t8_forest/t8_forest_partition.cxx (hunk 4 above)
Comment (Prasanna-Ponnusamy) [reply src/t8_forest/t8_forest_partition.cxx]: no, wasn't intentional. deleted

File: src/t8_data/t8_stdvector_conversion.hxx (hunk 3 above)
Comment (Prasanna-Ponnusamy) [reply src/t8_data/t8_stdvector_conversion.hxx:43]: hope it is clear now. If not please let me know.

**Comment by Davknapp** [comment]
Can you update the indentation please?

**Pull request closed without merging**